diff --git a/ihp-sg13g2/libs.tech/klayout/tech/lvs/sg13g2.lvs b/ihp-sg13g2/libs.tech/klayout/tech/lvs/sg13g2.lvs
index e33911b..08c2ed2 100644
--- a/ihp-sg13g2/libs.tech/klayout/tech/lvs/sg13g2.lvs
+++ b/ihp-sg13g2/libs.tech/klayout/tech/lvs/sg13g2.lvs
@@ -296,6 +296,33 @@ logger.info('Starting base layers derivations')
 
 # %include rule_decks/tap_derivations.lvs
 
+#================================================
+#------ FlowSpace SRAM custom connection --------
+#================================================
+
+logger.info('FlowSpace custom setup')
+
+# *Precharge cell does not have nwell tap but has pmos
+# *Cell cell does not have nwell or bulk tap but has nmos and pmos
+# In SRAM all NWell will in the end be connected to 'vdd' net and
+# pwell to 'vss'
+# This will break LVS on circuits that have NWell on different nets.
+# connect_global(nwell_drw, 'vdd')
+
+# vdd may be different unconnected nets in the sub-cells
+# see also https://github.com/IHP-GmbH/IHP-Open-PDK/issues/364
+# We assume all SRAM sub-cells start with SP6T or DP8T
+connect_implicit('SP6T*', 'vdd')
+connect_implicit('DP8T*', 'vdd')
+
+# Some hacking is needed to handle shared source/drain regions
+# between different cells
+sharedsd_cells = [
+  "SP6TCell", "SP6TPrecharge", "SP6TRowDecoderNand3X2", "SP6TWLDrive_26LN100WN26LP200WP",
+  "DP8TCell", "DP8TPrecharge_bl1", "DP8TPrecharge_bl2", "DP8TRowDecoderNand3X2",
+  "DP8TWLDrive_26LN100WN26LP200WP_wl1", "DP8TWLDrive_26LN100WN26LP200WP_wl2"
+]
+
 #================================================
 #------------ DEVICES CONNECTIVITY --------------
 #================================================
@@ -312,7 +339,9 @@ logger.info('Starting SG13G2 LVS DEVICES EXTRACTION')
 # ----- MOSFET EXTRACTION -------
 #================================
 
+cheat(sharedsd_cells) {
 # %include rule_decks/mos_extraction.lvs
+}
 
 #================================
 # ---- RF-MOSFET EXTRACTION -----
